tools/devops/automation/scripts/bash/build-mlaunch-nuget.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -ex
+
+# env var should have been defined by the CI
+if test -z "$XAM_TOP"; then
+    echo "Variable XAM_TOP is missing."
+    exit 1
+fi
+
+if test -z "$MACCORE_TOP"; then
+    echo "Variable MACCORE_TOP is missing."
+    exit 1
+fi
+
+cd "$XAM_TOP"
+
+MACCORE_HASH=$(cd "$MACCORE_TOP" && git log -1 --pretty=%h)
+
+if nuget list -source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json -AllVersions -Prerelease Microsoft.DotNet.Mlaunch | grep $MACCORE_HASH; then
+    echo "Mlaunch revision $MACCORE_HASH is already published as nupkg"
+    exit 0
+fi
+
+# Package mlaunch as .nupkg
+echo "Packaging mlaunch revision $MACCORE_HASH as nupkg..."
+
+DOTNET_NUPKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_NUPKG_DIR | grep "^DOTNET_NUPKG_DIR=" | sed -e 's/^DOTNET_NUPKG_DIR=//')
+MLAUNCH_WORK_DIR="$DOTNET_NUPKG_DIR/mlaunch-staging"
+
+rm -rf "$MLAUNCH_WORK_DIR"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/bin"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+
+DOTNET6=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET6 | grep "^DOTNET6=" | sed -e 's/^DOTNET6=//')
+IOS_DESTDIR=$(make -C tools/devops print-abspath-variable VARIABLE=IOS_DESTDIR | grep "^IOS_DESTDIR=" | sed -e 's/^IOS_DESTDIR=//')
+MONOTOUCH_PREFIX=$(make -C tools/devops print-abspath-variable VARIABLE=MONOTOUCH_PREFIX | grep "^MONOTOUCH_PREFIX=" | sed -e 's/^MONOTOUCH_PREFIX=//')
+
+# Copy mlaunch to staging area
+cp -c -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+cp -c "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
+
+# Add the .csproj we will use to create the .nupkg
+cp -c "$XAM_TOP/tools/mlaunch/Microsoft.DotNet.Mlaunch.csproj" "$MLAUNCH_WORK_DIR"
+
+# We need to override global.json to use .NET 6.0
+cp -c "$XAM_TOP/global6.json" "$MLAUNCH_WORK_DIR/global.json"
+
+version=$("$MLAUNCH_WORK_DIR/mlaunch/bin/mlaunch" --version | cut -d " " -f 2)
```

Comment:
Let's do what Sebastien said, and use the Xcode version (the XCODE_VERSION variable in Make.config).

Reply:
Done!